### PR TITLE
feat: Create teams component module

### PR DESCRIPTION
**Pull request by jai**
Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

Create PagerDuty Teams component module in order to update our Incident Management
infrastructure.

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-pagerduty-alerting/12)

### CODE REVIEW

**Comment by jai** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12** <a href="https://app.graphite.dev/github/pr/honestbank/terraform-pagerduty/12" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/honestbank/terraform-pagerduty/12?utm_source=stack-comment).

**Review by poom — COMMENTED** [review]
Something wrong for this file

`examples/pagerduty-team/README.md`

It should be generated readme

File: modules/pagerduty-team/inputs.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+variable "description" {
+  description = "A description of the team. If not set, a placeholder of `Managed by Terraform` will be set."
+  type        = string
+}
+
+variable "name" {
+  description = "The name of the team."
+  type        = string
+}
```
Comment (poom) [inline modules/pagerduty-team/inputs.tf:9]: do you want to create subteam? https://registry.terraform.io/providers/PagerDuty/pagerduty/latest/docs/resources/team#parent

Comment (poom) [reply modules/pagerduty-team/inputs.tf:9]: and I think we can add membership modules (3 modules) to this module too

memberships_observer = [user_id1,user_id1]
memberships_responder = [user_id1,user_id1]
memberships_manager = [user_id1,user_id1]

Comment (JoseFMP) [reply modules/pagerduty-team/inputs.tf:9]: I added now input for members. I think no need to have three separate inputs. Just one `members` is enough.

**Review by JoseFMP — APPROVED** [review]

**Review by poom — APPROVED** [review]

**Pull request merged**